source/platform_apple.cmake

```diff
@@ -0,0 +1,48 @@
+if( NOT APPLE)
+    message(FATAL_ERROR "Don't include this on non-macOS platforms.")
+endif()
+
+include(CheckIncludeFileCXX)
+check_include_file_cxx(filesystem HAVE_FILESYSTEM)
+
+if(NOT HAVE_FILESYSTEM)
+  message(FATAL_ERROR "filesystem header NOT found. Please use LLVM Clang or upgrade to Xcode 11.")
+endif()
+
+# Darwin 19 (macOS 10.15) adds the std::filesystem::path.
+if(CMAKE_SYSTEM_VERSION VERSION_LESS 19)
+    # This fixes the "path explicitly marked unavailable here" error.
+    add_definitions("-D_LIBCPP_DISABLE_AVAILABILITY")
```

Comment:
I have not found this macro on the [latest Clang documentation](https://libcxx.llvm.org/docs/UsingLibcxx.html). Can you put a link into the comment where is it from?

Reply:
Added link to it in explanatory comment. https://github.com/llvm/llvm-project/blob/2eadbc86142bab5b46dfeb55d8bd6724234278bb/libcxx/include/__availability#L19